pkg/apis/agones/v1/gameserver.go

```diff
@@ -715,18 +715,33 @@ func (gs *GameServer) Pod(apiHooks APIHooks, sidecars ...corev1.Container) (*cor
 
 	gs.podObjectMeta(pod)
 	for _, p := range gs.Spec.Ports {
-		cp := corev1.ContainerPort{
-			ContainerPort: p.ContainerPort,
-			HostPort:      p.HostPort,
-			Protocol:      p.Protocol,
-		}
-		err := gs.ApplyToPodContainer(pod, *p.Container, func(c corev1.Container) corev1.Container {
-			c.Ports = append(c.Ports, cp)
-
-			return c
-		})
-		if err != nil {
-			return nil, err
+		if p.PortPolicy == Static && p.Protocol == ProtocolTCPUDP {
+			// Handle TCPUDP ports by duplicating the port entry for both TCP and UDP for Static
+			cpTCP := corev1.ContainerPort{
+				ContainerPort: p.ContainerPort,
+				HostPort:      p.HostPort,
+				Protocol:      corev1.ProtocolTCP,
+			}
+			cpUDP := corev1.ContainerPort{
+				ContainerPort: p.ContainerPort,
+				HostPort:      p.HostPort,
+				Protocol:      corev1.ProtocolUDP,
+			}
+			pod.Spec.Containers[0].Ports = append(pod.Spec.Containers[0].Ports, cpTCP, cpUDP)
+		} else {
+			cp := corev1.ContainerPort{
+				ContainerPort: p.ContainerPort,
+				HostPort:      p.HostPort,
+				Protocol:      p.Protocol,
+			}
+			err := gs.ApplyToPodContainer(pod, *p.Container, func(c corev1.Container) corev1.Container {
+				c.Ports = append(c.Ports, cp)
+
+				return c
+			})
+			if err != nil {
+				return nil, err
+			}
```

Comment:
So  my suggestion would be to create a unit test to make it easier to test if the output of this function matches what you expect.

Here is an example: https://github.com/googleforgames/agones/blob/30f247dd3e276fe14fe87ed43efcc639ed839902/pkg/apis/agones/v1/gameserver_test.go#L1240

Reply:
Thanks! I added the test cases.